[new] Migrate to MST (#1346)
timescale/docs: mst/migrate-to-mst.md

---

> title: Migrating an existing TimescaleDB database to Managed Service for TimescaleDB

Reviewer: ```suggestion
title: Migrating from self-hosted TimescaleDB to Managed Service for TimescaleDB
```

excerpt: Migrating an existing TimescaleDB database to Managed Service for TimescaleDB
product: mst

> keywords: [migrate, databse]

Reviewer: ```suggestion
keywords: [migrate, database]
```

tags: [ingest, backup, restore]
---

# Migrate the entire database to Managed Service for TimescaleDB

You can migrate your data from self-hosted TimescaleDB to Managed Service for
TimescaleDB and use exclusive features, including separate scaling for compute

> and storage requirements, first-class multi-node support, and more.

Reviewer: Separate scaling and multi-node are Timescale Cloud-exclusive features. As far as I'm aware, they're not available on MST.

Author: > Separate scaling and multi-node are Timescale Cloud-exclusive features. As far as I'm aware, they're not available on MST.

Let me confirm, this as well.

Author: Hi @alexthinath  could you please let me know the features that are exclusive to MST.

Reviewer 2: yes Charis's comment is correct. To be a bit more specific, on Cloud you can choose Storage and Compute completely independently. There is a slider bar for each resource. On MST, you can only choose a "plan" which is a combination of CPU+Storage. E.g. 100GB storage + 2 CPU. From there you can size up the disk amount, but you're still choosing a "base plan" that couples the two resources.

Managed Service for TimescaleDB creates a database named `defaultdb` and a
default user account named `tsdbadmin`. You can use the Web console to create
additional users and databases using the `Users` and `Databases` tabs. 

You can switch between different plan sizes in Managed Service for TimescaleDB
but for the duration of the dumping process choose a plan size that is large

> enough for the task. This allows you to limit the downtime during the migration

Reviewer: Do we have a suggestion for what plan size is "large enough"?

Author: > Do we have a suggestion for what plan size is "large enough"?

good question. Let me find out.

Author: @alexthinath  could you please let know what plan size is "large enough"?

Reviewer 2: Perhaps we update the text to say something like "the same storage size or slightly larger than the currently allocated plan"

process and have a sufficiently powerful TimescaleDB plan.

<highlight type="warning">
Depending on your database size and network speed, migration can take a very
long time. During this time, any new writes that happen during the dumping

> process are not included, hence to get all your data you should turn off all the

Reviewer: ```suggestion
process are not included. To prevent data loss, turn off all the
```

writes to the old database server before you start the dumping process. Try to
migrate as a cold run without turning off writes on your previous server and
running the dump concurrently. This gives you an estimate of the time the dump
process takes. It also helps you in practicing the actual operation when it's
not causing downtime to your customers.
</highlight>

## Before you begin
Ensure that you have:
*   Installed the PostgreSQL [`pg_dump`][pg_dump] and [`pg_restore`][pg_restore]
    utilities.
*   Installed a client for connecting to PostgreSQL. These instructions use
    [`psql`][psql], but any client works.
*   Created a new empty database in Managed Service for TimescaleDB. For more
    information, see the [Install Managed Service for TimescaleDB][install-mst].
    Provision your database with enough space for all your data.
*   Checked that you're running the same major version of PostgreSQL on both
    Managed Service for TimescaleDB and your source database. For information

>     about upgrading PostgreSQL on your source database, see the [upgrade

Reviewer: ```suggestion
    about upgrading PostgreSQL, see the [upgrade
```
